### PR TITLE
Fix ClusterBasedProcessorLifecycleListener test to exit without waiting for timeout

### PR DESCRIPTION
**Pull request by abhishekshivanna**

### CODE REVIEW

File: samza-core/src/test/java/org/apache/samza/runtime/TestClusterBasedProcessorLifecycleListener.java [hunk 1]
```diff
@@ -79,11 +79,14 @@ public void testLifecycleListenerAfterFailure() {
...
 
-    // setup shutdown hook
+    // call beforeStart to setup shutdownHook
     clusterBasedProcessorLifecycleListener.beforeStart();
 
     // Simulating shutdown hook invocation by JVM
     clusterBasedProcessorLifecycleListener.getShutdownHookThread().run();
+
+    // call afterStop to countdown shutdownLatch
+    clusterBasedProcessorLifecycleListener.afterStop();
```
Comment (bkonold) [inline samza-core/src/test/java/org/apache/samza/runtime/TestClusterBasedProcessorLifecycleListener.java]: Just to clarify, this is to address the following line always waiting until the timeout? https://github.com/apache/samza/blob/9b3a9e68bca8cfef8a87542143212015b5263cea/samza-core/src/main/java/org/apache/samza/runtime/ClusterBasedProcessorLifecycleListener.java#L95

File: samza-core/src/test/java/org/apache/samza/runtime/TestClusterBasedProcessorLifecycleListener.java [hunk 2]
```diff
@@ -79,11 +79,14 @@ public void testLifecycleListenerAfterFailure() {
   public void testShutdownHookInvokesShutdownHookCallback() {
     doNothing().when(mockShutdownHookCallback).run();
 
-    // setup shutdown hook
+    // call beforeStart to setup shutdownHook
```
Comment (cameronlee314) [inline samza-core/src/test/java/org/apache/samza/runtime/TestClusterBasedProcessorLifecycleListener.java]: I think the timeout is actually set to 1ms in the setup, so this still might trigger the timeout. Could you set the timeout to something bigger? Then at least some could manually see the log to verify that the timeout did not get hit.

File: samza-core/src/test/java/org/apache/samza/runtime/TestClusterBasedProcessorLifecycleListener.java (hunk 1 above)
Comment (abhishekshivanna) [reply samza-core/src/test/java/org/apache/samza/runtime/TestClusterBasedProcessorLifecycleListener.java]: Yes, that is correct. Sorry I didn't make it clear.

File: samza-core/src/test/java/org/apache/samza/runtime/TestClusterBasedProcessorLifecycleListener.java (hunk 2 above)
Comment (abhishekshivanna) [reply samza-core/src/test/java/org/apache/samza/runtime/TestClusterBasedProcessorLifecycleListener.java]: Ah your right ! I moved the `afterStop()` into the callback so we can verify the invocation.

**Pull request merged**